client/landing/gutenboarding/index.tsx

```diff
@@ -2,37 +2,106 @@
  * External dependencies
  */
 import '@automattic/calypso-polyfills';
+import { I18nProvider } from '@automattic/react-i18n';
+import { getLanguageFile } from '../../lib/i18n-utils/switch-locale';
 import React from 'react';
 import ReactDom from 'react-dom';
 import { BrowserRouter } from 'react-router-dom';
 import config from '../../config';
+import { subscribe, select } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import { Gutenboard } from './gutenboard';
 import { setupWpDataDebug } from './devtools';
 import accessibleFocus from 'lib/accessible-focus';
+import { USER_STORE } from './stores/user';
+
 /**
  * Style dependencies
  */
 import 'assets/stylesheets/gutenboarding.scss';
 import 'components/environment-badge/style.scss';
 
-window.AppBoot = () => {
+const DEFAULT_LOCALE_SLUG: string = config( 'i18n_default_locale_slug' );
+
+interface AppWindow extends Window {
+	currentUser?: import('@automattic/data-stores').User.CurrentUser;
+	i18nLocaleStrings?: string;
+}
+declare const window: AppWindow;
+
+window.AppBoot = async () => {
 	if ( ! config.isEnabled( 'gutenboarding' ) ) {
 		window.location.href = '/';
-	} else {
-		setupWpDataDebug();
+		return;
+	}
+	setupWpDataDebug();
 
-		// Add accessible-focus listener.
-		accessibleFocus();
+	// Add accessible-focus listener.
+	accessibleFocus();
 
-		ReactDom.render(
+	let locale, localeData;
+	try {
+		[ locale, localeData ] = await getLocale();
+	} catch {
+		locale = DEFAULT_LOCALE_SLUG;
+		localeData = {};
+	}
+
+	ReactDom.render(
+		<I18nProvider locale={ locale } localeData={ localeData }>
 			<BrowserRouter basename="gutenboarding">
 				<Gutenboard />
-			</BrowserRouter>,
-			document.getElementById( 'wpcom' )
-		);
-	}
+			</BrowserRouter>
+		</I18nProvider>,
+		document.getElementById( 'wpcom' )
+	);
 };
+
+/**
+ * Load the user's locale
+ *
+ * 1. If i18nLocalStrings is present use those strings and data.
+ * 2. If we have a currentUser object, use that locale to fetch data.
+ * 3. Fetch the current user and use language to fetch data.
+ * 4. TODO (#39312): If we have a URL locale slug, fetch and use data.
+ * 5. Fallback to "en" locale without data.
+ *
+ * @returns Tuple of locale slug and locale data
+ */
+async function getLocale(): Promise< [ string, object ] > {
```

Comment:
This is roughly analagous to `setupLocale` from [`client/boot/locale`](https://github.com/Automattic/wp-calypso/blob/master/client/boot/locale.js) with a few differences:

- No global Redux state here.
- No `lib/user` User eventEmitter object here.
- No `i18n-calypso` here.
- A few pieces are left for follow-ups if deemed necessary. See TODO in description.